src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -27,9 +17,39 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // implement new recipe
+    String recipeName = request.getParameter("recipeName");
+    String tagsResponse = request.getParameter("tags");
+    String ingredientsResponse = request.getParameter("ingredients");
+    String stepsResponse = request.getParameter("steps");
+    String privacy = request.getParameter("privacy");
 
-    response.sendRedirect("/pages/UserPage.html");
-  }
+    List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.split(",", 0)));
+    List<String> ingredients = new ArrayList<String>(Arrays.asList(ingredientsResponse.split("\n", 0)));
```

Comment:
Some comments here and below to explain what you're doing and why would help in the future.